plugins/CoreUpdater/Diagnostic/HttpsUpdateCheck.php

```diff
@@ -31,7 +32,7 @@ public function execute()
     {
         $label = $this->translator->translate('Installation_SystemCheckUpdateHttps');
 
-        if (CoreUpdater\Controller::isUpdatingOverHttps()) {
+        if (Http::isUpdatingOverHttps() || GeneralConfig::getConfigValue('force_matomo_ssl_request') === 0) {
             return array(DiagnosticResult::singleResult($label, DiagnosticResult::STATUS_OK));
         }
```

Comment:
The logic should be something like the below (the statuses and `$comment`s are placeholders):
```suggestion
        if (GeneralConfig::getConfigValue('force_matomo_ssl_request') === 0) {
            $comment = 'An info message';
            return array(DiagnosticResult::singleResult($label, DiagnosticResult::STATUS_INFORMATIONAL, $comment));
        } elseif (Http::isUpdatingOverHttps()) {
            return array(DiagnosticResult::singleResult($label, DiagnosticResult::STATUS_OK));
        } else {
            $comment = 'An error message';
            return array(DiagnosticResult::singleResult($label, DiagnosticResult::STATUS_ERROR, $comment));
        }
```